src/transformers/pipelines/pt_utils.py

```diff
@@ -111,6 +111,44 @@ def loader_batch_item(self):
             # Recreate the element by reusing the original class to make it look
             # batch_size=1
             result = self._loader_batch_data.__class__(loader_batched)
+        else:
+            # Batch data is assumed to be tuple
+            loader_batched = []
+            for element in self._loader_batch_data:
+                if isinstance(element, ModelOutput):
+                    # Convert ModelOutput to tuple first
+                    element = element.to_tuple()
+                    if isinstance(element[0], torch.Tensor):
+                        loader_batched.append(tuple(el[self._loader_batch_index].unsqueeze(0) for el in element))
+                    elif isinstance(element[0], np.ndarray):
+                        loader_batched.append(tuple(np.expand_dims(el[self._loader_batch_index], 0) for el in element))
+                    continue
+                if isinstance(element, tuple):
```

Comment:
This isn't equivalent to when `return_dict=True` as this will enter here for all values in the outputs rather than just for `"hidden_states", `"past_key_values"` and `"attentions"`.